diff --git a/code/nnv/engine/nn/layers/FeatureInputLayer.m b/code/nnv/engine/nn/layers/FeatureInputLayer.m
index 14ad8a5d2..2822f1094 100644
--- a/code/nnv/engine/nn/layers/FeatureInputLayer.m
+++ b/code/nnv/engine/nn/layers/FeatureInputLayer.m
@@ -94,17 +94,16 @@ classdef FeatureInputLayer < handle
         function out_set = reach_star_single_input(obj, in_set)
             % @in_set: an input ImageStar
             % @out_set: an output ImageStar
-            n = in_set.dim;
-            
+
             % Compute normalization
             if strcmp(obj.Normalization, 'none')
                 out_set = in_set;
             elseif strcmp(obj.Normalization, 'zerocenter')
-                out_set = in_set.affineMap(eye(n), -obj.Mean);
+                out_set = in_set.affineMap([], -obj.Mean);
             else
                 error('The normalization method is not supported yet.')
             end
-            
+
         end
         
         % handling multiple inputs (star)
